tools/comdb2ar/serialise.cpp

```diff
@@ -893,36 +950,122 @@ void serialise_database(
 
     // Construct a manifest which will give the page sizes of all the files
     std::ostringstream manifest;
-    manifest << "# Manifest for serialisation of " << dbname << std::endl;
-    if(support_files_only) {
-        manifest << "SupportFilesOnly" << std::endl;
-        if (origlrlname != "")
-        {
-           manifest <<"OrigLrlFile "<< origlrlname<< std::endl;
+
+    // Non-incremental mode or increment creation mode
+    if(!incr_gen){
+        manifest << "# Manifest for serialisation of " << dbname << std::endl;
+        if(support_files_only) {
+            manifest << "SupportFilesOnly" << std::endl;
+            if (origlrlname != "")
+            {
+               manifest <<"OrigLrlFile "<< origlrlname<< std::endl;
+            }
         }
-    }
 
-    for(std::list<FileInfo>::const_iterator
-            it = data_files.begin();
-            it != data_files.end();
-            ++it) {
-            write_manifest_entry(manifest, *it);
-    }
+        for(std::list<FileInfo>::const_iterator
+                it = data_files.begin();
+                it != data_files.end();
+                ++it) {
+                write_manifest_entry(manifest, *it);
+        }
+
+        // Find a recovery point after the copy, and record it in the manifest
+        if (!support_files_only) {
+            std::clog << "logdelete version " << log_holder->version() << std::endl;
+            if (log_holder->version() >= 3) {
+                std::string recovery_options = log_holder->recovery_options();
+                if (!recovery_options.empty()) {
+                    manifest << "Option " << recovery_options <<std::endl;
+                }
+            }
+        }
+
+        // Serialise the manifest file
+        serialise_string("MANIFEST", manifest.str());
+
+    // Incremental Mode
+    } else {
+        manifest << "# Manifest for serialisation of increment produced on "
+            << getDTString() << std::endl;
+
+
```

Comment:
The "logdelete version" / "recovery_options" logic above should apply to incremental backups as well.